ci/e2e.go

```diff
@@ -49,6 +63,39 @@ func (m *Ci) Etoe(
 		Exec(args)
 }
 
+func (e *Etoe) Diff(
+	ctx context.Context,
+	// The base ref against which to run the diffs
+	baseRef string,
+) error {
+	db := postgres()
+
+	// Migrate DB with base state
+	_, err := goModule().
+		WithCgoEnabled().
+		WithSource(e.Ci.Source).
+		Container().
+		WithExec([]string{"apk", "add", "--update", "--no-cache", "ca-certificates", "make", "git", "curl", "clang", "lld"}).
+		WithExec([]string{"git", "checkout", "-f", baseRef}).
+		WithServiceBinding("postgres", db).
+		WithEnvVariable("POSTGRES_URL", "postgres://postgres:postgres@postgres:5432/postgres").
+		WithExec([]string{"go", "run", "./tools/migrate"}).
+		Sync(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to migrate base ref %s: %w", baseRef, err)
+	}
+
+	_, err = goModule().
+		WithSource(e.Ci.Source).
+		Container().
+		WithServiceBinding("postgres", db).
+		WithEnvVariable("POSTGRES_URL", "postgres://postgres:postgres@postgres:5432/postgres").
```

Comment:
These could be consolidated into a single container with the above. It would improve caching.

Reply:
consolidated them to single container

Reply:
WithExec doesn't provide me a proper shell so I couldn't find a way to go back to the original revision (i couldn't get the current head sha to env or an external variable) without using a hack like a local tag i name or a util script, so i separated them again

Reply:
You can `WithExec([]string{"sh", "-c", "git something something"})

Reply:
Let's chat about this sometime. Happy to help you get this sorted.

Reply:
What I meant is rather that I could only do it conceptually similar to this
```go
// Stash local changes
WithExec([]string{"git", "stash", "save", "-u", "-m", "diffteststash"}).
WithExec([]string{"ash", "-c", "echo $(git rev-parse HEAD) > /tmp/revision"}).
// Migrate
WithExec([]string{"git", "checkout", "-f", baseRef}).
WithExec([]string{"go", "run", "./tools/migrate"}).
// Return to original version
WithExec([]string{"ash", "-c", "git checkout $(cat /tmp/revision)"}).
WithExec([]string{"ash", "-c", "git stash apply stash^diffteststash || true"}).
// Run tests
WithExec([]string{"go", "test", "-v", "./e2e/diff/..."}).
WithExec([]string{"git", "stash", "apply"}).
```
which i consider to be hacky. I don't know of a way to get the revision value without using a workaround like this, that's why I decided to separate them again. (I wanted to write it to environment to extract it with `.EnvVariable("x")`, but the string arglist is not a proper shell and when using a subshell i don't have access to the parent env...)

(If a similar approach doesn't bother you then we could use it, but I'd personally rather just have two containers)